apps/foundation-demo/src/app/design/molecules/sidebar/sidebar/sidebar.component.spec.ts

```diff
@@ -0,0 +1,104 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+
+import { DaffSidebarComponent } from './sidebar.component';
+import { Component } from '@angular/core';
+import { By } from '@angular/platform-browser';
+
+describe('DaffSidebarComponent', () => {
+  let component: DaffSidebarComponent;
+  let fixture: ComponentFixture<DaffSidebarComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ 
+        DaffSidebarComponent
+      ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(DaffSidebarComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  })
+  
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('user interactions', () => {
+    it('should emit `escapedPressed` when the `ESC` key is pressed', () => {
+      spyOn(component.escapePressed, "emit");
+
+      fixture.nativeElement.dispatchEvent(new KeyboardEvent("keydown", {
+          key: "Escape"
+      }));
+      
+      fixture.detectChanges();
+
+      expect(component.escapePressed.emit).toHaveBeenCalled();
+    });
+
+    it('should not emit `escapedPressed` if the event is not triggered ON the `daff-sidebar`', () => {
+      spyOn(component.escapePressed, "emit");
+
+      document.dispatchEvent(new KeyboardEvent("keydown", {
+        key: "Escape"
+      }));
+
+      fixture.detectChanges();
+
+      expect(component.escapePressed.emit).not.toHaveBeenCalled();
+    });
+  });
+});
+
+@Component({template: `
+  <div class="host-element">
+    <daff-sidebar (escapePressed)="pressed()"></daff-sidebar>
+  </div>
+`})
+class TestDaffSidebarComponentWrapper {
+  escapePressedCount: number = 0;
+
+  pressed(): void{
+    this.escapePressedCount++;
+  }
+}
+
+describe('DaffSidebarComponent | usage', () => {
```

Comment:
can't both outer describes be under one DaffSidebarComponent describe?

Reply:
So, my thoughts here was that I figured it'd be clearer to test just the internals succinctly own describe and then how a client would consume the component in another test (potentially in another spec file located nearby).